record/record.go

```diff
@@ -127,11 +131,40 @@ type Reader struct {
 
 // NewReader returns a new reader.
 func NewReader(r io.Reader) *Reader {
+	rs, _ := r.(io.ReadSeeker)
 	return &Reader{
-		r: r,
+		r:  r,
+		rs: rs,
 	}
 }
 
+// Skips all blocks that are completely before offset.
+// Returns the new offset in the file if successful, error otherwise.
+func (r *Reader) SkipToInitialOffset(offset int64) (int64, error) {
```

Comment:
It's totally a C++ name, so my apologies on that front.  As I explained above, it's a holdover from how this was done in C++, and I'm not terribly a fan of having some magical initial offset, when we could potentially allow arbitrary seeking.

I have a few thoughts on this...

Logically, we should only ever Seek to a 32KB aligned offset, if the offset is not 32KB aligned we can either coerce it to be 32KB aligned or return an error.  I think coercing it to be 32KB aligned is easiest and ideal. Since Seek returns the new offset, there is no obligation that the returned offset = the existing offset + the requested value.

Regardless of that, the following call to Next will set wantFirst = true, and then we'll resync with a valid FIRST or FULL chunk next time it's called. Note that Seek, if implemented by us, will have to clear r.{i,j,n} to trigger a new buf copy on Next's next execution.

Or, as a final alternative, we could guard the following line:  n, err := io.ReadFull(r.r, r.buf[:]) with a check that r.r is 32KB aligned.  Then we'll actively enforce never reading from an unaligned offset and we sidestep the whole issue - allowing Seeking to occur anywhere, and if we seek outside of the current buffer, we leave it to the caller entirely to Recover.

I think creating a Seek method that accepts any input and returns the offset of the next 32KB aligned block, is the best way of doing it. The next FIRST or FULL record might not occur at that exact offset, but it is the offset from which we'll start looking for a new record.

The only requirement to make this good would be having NewReader, accept an io.ReadSeeker instead of io.Reader.  Or I could leave it as it is, and test if the provided io.Reader is an io.ReadSeeker and seeking only when possible.